### PR TITLE
feat: add support for nested spans

### PR DESCRIPTION
**Pull request by nicolasiscoding**

### CODE REVIEW

File: src/helpers/xml-builder.js [hunk 1]
```diff
@@ -364,6 +369,35 @@ const buildRun = async (vNode, attributes, docxDocumentInstance) => {
...
+            array.forEach((fragmentFromSpan) => {
+              if (Array.isArray(fragmentFromSpan)) {
+                dfs(fragmentFromSpan);
+              } else {
+                runFragmentsArray.push(fragmentFromSpan);
+              }
+            });
+          };
+
+          dfs(spanFragment);
```
Comment (privateOmega) [inline src/helpers/xml-builder.js:393]:
```js
spanFragment.flat(Infinity);
runFragmentsArray.push(...spanFragment);
```
 should do the same trick I guess

**Review by privateOmega — CHANGES_REQUESTED** [review]

File: example/example-node.js [hunk 2]
```diff
@@ -78,6 +78,12 @@ const htmlString = `<!DOCTYPE html>
             <li>Ordered list element</li>
         </ol>
         <div class="page-break" style="page-break-after: always"></div>
+        <ul>
+            <li><span style="font-size:10pt"><span style="color:##e28743">I am a teacup <strong><span style="color:#595959">A strong teacup</span></strong> Managed Environment.</span></span></li>
```
[the remark this answers is not in the document]
Comment (nicolasiscoding) [reply example/example-node.js]: Done

File: src/helpers/xml-builder.js (hunk 1 above)
Comment (nicolasiscoding) [reply src/helpers/xml-builder.js]: Done

File: src/helpers/xml-builder.js [hunk 3]
```diff
@@ -364,6 +369,35 @@ const buildRun = async (vNode, attributes, docxDocumentInstance) => {
         }
         const formattingFragment = buildTextFormatting(tempVNode);
         runPropertiesFragment.import(formattingFragment);
+        // go a layer deeper if there is a span somewhere in the children
+      } else if (isVNode(tempVNode) && tempVNode.tagName === 'span') {
```
[the remark this answers is not in the document]
Comment (privateOmega) [reply src/helpers/xml-builder.js:373]: Since isVNode is common in 2 if conditions, make it nested

[the remark this answers is not in the document]
Comment (nicolasiscoding) [reply src/helpers/xml-builder.js]: Done

**Comment by privateOmega** [comment]
@nicolasiscoding Please rebase this branch

**Comment by nicolasiscoding** [comment]
Rebased but did not merge develop into my branch

**Review by privateOmega — APPROVED** [review]
LGTM

**Pull request merged**